hail/python/test/hail/matrixtable/test_matrix_table_from_parts.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import hail as hl
+
+class Inits():
+    globals={'hello': 'world'}
+    rows={'foo': ['a', 'b']}
+    cols={'bar': ['c', 'd']}
+    entries={'baz': [[1, 2], [3, 4]]}
+
+def unless(test: bool, kvs):
+    return {} if test else kvs
+
+def assert_matches_globals(mt: 'hl.MatrixTable', no_props=False):
+    assert hl.eval(mt.globals) == hl.Struct(**unless(no_props, Inits.globals))
+
+def assert_matches_rows(mt: 'hl.MatrixTable', no_props=False):
+    assert mt.rows().order_by('row_idx').collect() == \
+        [ hl.Struct(row_idx=0, **unless(no_props, {'foo':'a'}))
+        , hl.Struct(row_idx=1, **unless(no_props, {'foo':'b'}))
+        ]
+
+def assert_matches_cols(mt: 'hl.MatrixTable', no_props=False):
+    assert mt.cols().order_by('col_idx').collect() == \
+        [ hl.Struct(col_idx=0, **unless(no_props, {'bar': 'c'}))
+        , hl.Struct(col_idx=1, **unless(no_props, {'bar': 'd'}))
+        ]
+
+def assert_matches_entries( mt: 'hl.MatrixTable'
+                          , no_row_props=False
+                          , no_col_props=False
+                          , no_entry_props=False
+                          ):
+    assert mt.entries().order_by('row_idx', 'col_idx').collect() == \
+        [ hl.Struct( row_idx=0
+                   , **unless(no_row_props, {'foo':'a'})
+                   , col_idx=0
+                   , **unless(no_col_props, {'bar':'c'})
+                   , **unless(no_entry_props, {'baz':1})
+                   )
+        , hl.Struct( row_idx=0
+                   , **unless(no_row_props, {'foo':'a'})
+                   , col_idx=1
+                   , **unless(no_col_props, {'bar':'d'})
+                   , **unless(no_entry_props, {'baz':2})
+                   )
+        , hl.Struct( row_idx=1
+                   , **unless(no_row_props, {'foo':'b'})
+                   , col_idx=0
+                   , **unless(no_col_props, {'bar':'c'})
+                   , **unless(no_entry_props, {'baz':3})
+                   )
+        , hl.Struct( row_idx=1
+                   , **unless(no_col_props, {'bar':'d'})
+                   , col_idx=1
+                   , **unless(no_row_props, {'foo':'b'})
+                   , **unless(no_entry_props, {'baz':4})
+                   )
+        ]
+
+def test_from_parts():
+    mt = hl.MatrixTable.from_parts( globals=Inits.globals
+                                  , rows=Inits.rows
+                                  , cols=Inits.cols
+                                  , entries=Inits.entries
+                                  )
+    assert_matches_globals(mt)
```

Comment:
I'm dubious on the relative clarity of:
```
assert_matches_globals(mt)
```
compared to
```
assert hl.eval(mt.globals) == Inits.globals
```

The latter states clearly what the assertion is whereas the former sends me hunting to the definition to understand what's being asserted.

Is there an argument in favor of the former that I'm missing?

Reply:
Agreed, bad function naming. I'm trying to limit the uses of the global data so that if things need to be changed, fewer uses need to be changed.